File: 트리/최소신장트리/1922.py
def find_parent(x):
    if parent[x] != x:
        parent[x] = find_parent(parent[x])
    return parent[x]

def union_parent(v1, v2):
    v1 = find_parent(v1)
    v2 = find_parent(v2)
    if v1 < v2 :
        parent[v2] = v1
    else:
        parent[v1] = v2

N = int(input()) # 컴퓨터의 수 (노드 개수)
M = int(input()) # 연결할 수 있는 선의 수 (간선 개수)
parent = [0] * (N+1)

for i in range(1, N+1):
    parent[i] = i

edge = []
for _ in range(M):
    a, b, c = map(int, input().split()) # a와 b를 연결하는 비용이 c
    edge.append((c,a,b))

edge.sort()

result = 0
for i in range(M):
    cost, a, b = edge[i]
    if find_parent(a) != find_parent(b):
        union_parent(a, b)
        result += cost
print(result)

File: 트리/최소신장트리/1922.java
import java.util.*;

class Main{
    public static int[] parent;
    public static int result = 0;
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int m = sc.nextInt();
        ArrayList<Edge> edges = new ArrayList<>();

        for(int i=0; i<m; i++){
            int a = sc.nextInt();
            int b = sc.nextInt();
            int cost = sc.nextInt();
            edges.add(new Edge(cost, a, b));
        }

        Collections.sort(edges);
        parent = new int[n+1];
        for(int i=1; i<n+1; i++){
            parent[i] = i;
        }
        for(int i=0; i<m; i++){
            int a = edges.get(i).getNodeA();
            int b = edges.get(i).getNodeB();
            if(findParent(a) != findParent(b)){
                unionParent(a, b);
                result += edges.get(i).getDistance();
            }
        }
        System.out.println(result);
    }

    public static int findParent(int x){
        if(x == parent[x]) return x;
        return parent[x] = findParent(parent[x]);
    }

    public static void unionParent(int a, int b){
        a = findParent(a);
        b = findParent(b);
        if(a < b) parent[b] = a;
        else parent[a] = b;
    }
}

class Edge implements Comparable<Edge>{
    private int distance;
    private int nodeA;
    private int nodeB;

    public Edge(int distance, int nodeA, int nodeB){
        this.distance = distance;
        this.nodeA = nodeA;
        this.nodeB = nodeB;
    }

    public int getDistance(){
        return this.distance;
    }

    public int getNodeA(){
        return this.nodeA;
    }

    public int getNodeB(){
        return this.nodeB;
    }

    @Override
    public int compareTo(Edge other){
        if(this.distance < other.getDistance()){
            return -1;
        }
        return 1;
    }
}
